bytes/src/main/scala/io/iohk/ethereum/utils/ByteStringUtils.scala

```diff
@@ -61,9 +61,14 @@ object ByteStringUtils {
   }
 
   def concatByteStrings(elements: Iterator[ByteStringElement]): ByteString = {
-    val builder = new mutable.ArrayBuilder.ofByte
-    elements.foreach(el => builder.addAll(el.asByteArray))
-    ByteString(builder.result())
+    val elementList: List[ByteStringElement] = elements.toList
```

Comment:
I am sorry I was not precise enough when we discuss this.

When we migrated from 2.12 to 2.13 there was a problem that ByteString do not support all collections API like concatenation (because of removal of CanBuildFrom). So it was possible to write some utility that will concatenate `Array[byte]` that checks size of arrays and prepare final one with proper size.

I am afraid that when argument is iterator we need to convert `toList` so we loose perf benefits 😢

Reply:
I have removed `toList`, hopefully a `Seq` from VarArg is fast enough prepending elements.